mitodl/micromasters
# Add MITx Online Login UI + Dashboard Prompts

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #4994 

#### What's this PR do?
(Required)

#### How should this be manually tested?

- As an anonymous user, try to go to a login required page like `/dashboard`, you should get redirected to `/signin`
- Clicking "Sign In" or "Create Account" on the header should bring you to the signin page with only edx login showing
- Clicking on "Sign In" from a program page should bring you to the sign in page with some additional context indicating where you need to login with your program depending on which platforms the runs are configured (see screenshots)
- If you login as a learner who only has edx auth, you should see a page informing you that you need an mitx online account and that should be functional in linking a new or existing mitx online account to your logged in micromasters account.

#### Any background context you want to provide?
I haven't done anything on the react side intentionally to keep things simple - 

#### Screenshots (if appropriate)

If the user is logging in from a program that is in MITx Online:
![Screenshot 2021-08-31 at 17-25-23 MITx MicroMasters](https://user-images.githubusercontent.com/28598/131578891-7cb52cb1-82e8-4b4c-9a38-6456ded2c826.png)

If the user is logging in from a program that is only in edX:
![Screenshot 2021-08-30 at 23-29-26 MITx MicroMasters](https://user-images.githubusercontent.com/28598/131578895-0bf0de81-6ecd-448f-9c98-80436b0e11e6.png)

Generic, no-context login:
![Screenshot 2021-08-30 at 23-26-15 MITx MicroMasters](https://user-images.githubusercontent.com/28598/131578896-26dcf732-dba1-4f8f-bd3b-ba034ed151a6.png)


The user gets redirected to this if they login as a edx-only learner.

![Screenshot 2021-08-31 at 17-25-00 MITx MicroMasters](https://user-images.githubusercontent.com/28598/131578894-d31ad976-ed23-4788-9f6e-e92c4d1cc5e9.png)
